### ISSUE TITLE
exponentiating int256 by 0 or 1 results in runtime revert

### ISSUE THREAD

**Issue by charles-cooper** [vyperlang/vyper#2989]
### Version Information

* vyper Version (output of `vyper --version`): https://github.com/vyperlang/vyper/commit/be2c59a604070c75212dca90beb4c33fed908c8b
* Python Version (output of `python --version`): 3.8.10

### What's your issue about?

the following code results in a runtime revert:
```vyper
@external
def foo() -> int256:
  x: int256 = 2
  return x ** 0
```

**Comment by fubuloubu** [comment]
This should probably be an optimizer pass and reduce itself to `1` or `x`

**Comment by charles-cooper** [comment]
we do have optimizer rules for `exp <x> 1` and `exp <x> 0`, two things might interfere with that though, safepow rules and that `<x>` could have side effects (and note we are extremely, maybe overly conservative right now with our side effect detection rules)

https://github.com/vyperlang/vyper/blob/9a02a44d7b18ef3a618d2d1903283fd61aa71c88/vyper/ir/optimizer.py#L304-L314

**Comment by charles-cooper** [comment]
this is fixed in #2983 (test here: https://github.com/vyperlang/vyper/pull/2983/files#diff-08afe8356161d17a86aba4e8afda160c9cc61eed014309071c76b59884afe9dbR90-R100)
